src/data-access/repositories/mongo/mongo-entity-converter.ts

```diff
@@ -211,6 +210,7 @@ export class MongoEntityConverter {
       _rank: part.rank,
       isOnAir: part.isOnAir(),
       isNext: part.isNext(),
+      endState: part.getEndState() ?? undefined,
```

Comment:
Does `part.getEndState()` possibly return null?
if not, and it instead possibly returns undefined, then you are setting it to undefined anyways, making the null coalescing not needed.

Reply:
You are right, it's not needed. It's been updated.